src/components/JohnTheRipper/JohnTheRipper.tsx

```diff
@@ -1,117 +1,188 @@
-    "\n\nStep 6: Click crack to commence the tool's execution." +
-    "\n\nStep 7: View the output block below to view the results of the tool's execution.";
+import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile";
+import { RenderComponent } from "../UserGuide/UserGuide";
+import InstallationModal from "../InstallationModal/InstallationModal";
+import { checkAllCommandsAvailability } from "../../utils/CommandAvailability";
 
+/**
+ * Represents the form values for the AirbaseNG component.
+ */
 interface FormValuesType {
     filePath: string;
     hash: string;
     fileType: string;
     mode: string;
-    wordlist: string;
-    incrementorder: string;
+    wordList: string;
+    incrementOrder: string;
 }
 
-const fileTypes = ["zip", "rar", "raw"];
-const mode = ["incremental", "dictionary", "single"];
-const incrementorder = [
-    "ASCII",
-    "LM_ASCII",
-    "AlNum",
-    "Alpha",
-    "LowerNum",
-    "UpperNum",
-    "LowerSpace",
-    "Lower",
-    "Upper",
-    "Digits",
-    "LM_ASCII",
-];
-
+/**
+ * The JohnTheRipper component.
+ * @returns The JohnTheRipper component.
+ */
 const JohnTheRipper = () => {
-    const [loading, setLoading] = useState(false);
-    const [output, setOutput] = useState("");
-    const [selectedFileTypeOption, setSelectedFileTypeOption] = useState("");
-    const [selectedModeOption, setselectedModeOption] = useState("");
-    const [selectedIncrementOption, setselectedIncrementOption] = useState("");
-    const [pid, setPid] = useState("");
-
-    let form = useForm({
+    // Component state variables
+    const [loading, setLoading] = useState(false); // State variable to indicate loading state.
+    const [output, setOutput] = useState(""); // State variable to store the output of the command execution.
+    const [pid, setPid] = useState(""); // State variable to store the process ID of the command execution.
+    const [loadingModal, setLoadingModal] = useState(true); // State variable to indicate loading state of the modal.
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable that indicates if the modal is opened.
+    const [allowSave, setAllowSave] = useState(false); // State variable to allow saving the output to a file.
+    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved.
+
+    // Component constants
+    const [selectedFileTypeOption, setSelectedFileTypeOption] = useState(""); // State variable to store the selected file type.
+    const [selectedModeOption, setSelectedModeOption] = useState(""); // State variable to store the selected crack mode.
+    const [selectedIncrementOption, setSelectedIncrementOption] = useState(""); // State variable to store the selected increment order.
```

Comment:
These are all react state hooks as indicated by the "useState". Please move them into that appropriate section. 
```suggestion
```